tests/conversion

```diff
@@ -0,0 +1,264 @@
+#!/bin/sh
+set -eu
+
+# Source helper functions.
+. ./bin/helpers
+
+install_deps jq unzip
+
+# Install LXD
+install_lxd
+
+# Install go from Snap.
+snap install go --channel=latest/edge --classic
```

Comment:
Why pick the `latest/edge` channel for `go`? If the default channel works, I'd stick with it, please.

This seems to actually slow things down as it goes from 1.23 to 1.22.5:

```
go (edge) 1.23-148755a from Canonical** installed
go: downloading github.com/canonical/lxd v0.0.0-20240718114802-16be33533525
go: github.com/canonical/lxd@v0.0.0-20240718114802-16be33533525 requires go >= 1.22.4; switching to go1.22.5
go: downloading go1.22.5 (linux/amd64)
```

Reply:
This is because of the toolchain directive. Not a fan of it personally. Because then Go downloads its own version of Go, and doesn't use the system one.